sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuSemaphore.scala

```diff
@@ -309,6 +317,10 @@ private final class SemaphoreTaskInfo(val taskAttemptId: Long) extends Logging {
       semaphore.release(numPermits)
       hasSemaphore = false
       lastHeld = System.currentTimeMillis()
+      nvtxRange match {
```

Comment:
how about:
`nvtxRange.safeClose()` then make the implicit also apply to `Option[AutoCloseable]`? 

Else, you could do: 
`nvtxRange.foreach(_.close)`. I am not sure you need `safeClose`, as `Some(null)` is not possible.

Reply:
Oh, I'd also `nvtxRange = None` after this, just in case.

Reply:
Oh hmm. I thought `safeClose` would catch the exception if anything is thrown while closing, so that the next line could be executed properly. I wanted to do it because a failure to close an nvtxRange should be a trivial failure and should not cancel the query processing. But I realized `safeClose` actually does not eat the exception here. I will just change this to `nvtxRange.foreach(_.close())`. This `close` very unlikely throws anyway.